src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1299,14 +1299,18 @@ public static SqlConnectionStringBuilder CreateConnectionStringBuilder(Connectio
 
                 connectionBuilder.EnclaveAttestationUrl = connectionDetails.EnclaveAttestationUrl;
             }
-            if (connectionDetails.Encrypt.HasValue)
+            if (connectionDetails.Encrypt != null)
             {
-                connectionBuilder.Encrypt = connectionDetails.Encrypt.Value;
+                connectionBuilder.Encrypt = connectionDetails.Encrypt;
             }
             if (connectionDetails.TrustServerCertificate.HasValue)
             {
                 connectionBuilder.TrustServerCertificate = connectionDetails.TrustServerCertificate.Value;
             }
+            if (connectionDetails.HostNameInCertificate != null)

```

Comment:
Is this something we'll then have to bring over to any other service (Import/Data virt) that creates its own connection as well?